charts/nx-agents/templates/service.yaml

```diff
@@ -0,0 +1,24 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: nx-cloud-workflow-controller-service
+  namespace: {{ .Values.global.namespace | quote }}
+  labels:
+    app: nx-cloud-workflow-controller
+    {{- include "nxCloud.app.labels" . | indent 4 }}
+  {{- if .Values.controller.service.annotations }}
+  annotations:
+    {{- toYaml .Values.controller.service.annotations | nindent 4 }}
+  {{- end }}
+spec:
+  type: {{ .Values.controller.service.type }}
+  {{- if and .Values.controller.service.loadBalancerIP (eq .Values.controller.service.type "LoadBalancer")}}
+  loadBalancerIP: {{ .Values.controller.service.loadBalancerIP }}
```

Comment:
interesting - why do we have this option? is it so we can make sure the Load Balancer exposes this service externally without having to add a specific route to it?

Reply:
no we actually use this ourselves and it can be critical to exposing the controller internally. For GKE this will actually create an internal-only load balancer on the reserved IP. I actually have to review this again as I think an annotation is also needed.

Other providers (AWS, Azure) likely have similar internal-only LB functionality